jetty-core/jetty-http3/jetty-http3-server/src/main/java/org/eclipse/jetty/http3/server/internal/HttpStreamOverHTTP3.java

```diff
@@ -528,6 +528,12 @@ public void onIdleTimeout(TimeoutException failure, BiConsumer<Runnable, Boolean
 
     public Runnable onFailure(Throwable failure)
     {
+        try (AutoLock ignored = lock.lock())
+        {
+            if (chunk != null)
+                chunk.release();
+            chunk = Content.Chunk.from(failure, true);
+        }
```

Comment:
I'd like to see, if possible, the remembering of the failed chunk in the same method across the 4 `HttpStream` implementations.
H1 does it in `consumeAvailable`, H2 does not do it, H3 now does it in `onFailure()`.
Can we consolitate that remembering into a similar patter for all 4 (if possible)?

Reply:
Done.